### PR TITLE
Make functional tests output xml instead of html and use VSTest instead of MSTest

### PR DESCRIPTION
**Pull request by scottbommarito**
Changed the outputs of our test scripts to output `xml` instead of `html` and to use `VSTest` instead of `MSTest` (when possible).

Also added a script to determine the url of the gallery being tested using an Azure service principal.

### CODE REVIEW

File: tests/Scripts/Configure-FunctionalTests.ps1 [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	return $stagingUrl
+}
+
+if($env:Slot -eq "Production")
+{
+    $GalleryUrl = $env:SERVICEROOT
+}
+else
+{
+    # Use the Azure management certificate to find the url of the desired slot
```
Comment (shishirx34) [inline tests/Scripts/Configure-FunctionalTests.ps1]: nit: indentation

File: tests/Scripts/Configure-FunctionalTests.ps1 [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	$dict = New-Object "System.Collections.Generic.Dictionary``2[System.String,System.String]"
+	$dict.Add('x-ms-version', ' 2014-02-01')
```
Comment (shishirx34) [inline tests/Scripts/Configure-FunctionalTests.ps1]: Is this header needed?

File: tests/Scripts/Configure-FunctionalTests.ps1 [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	    $url = GetUrl "$response.Content"
+	    $url = $url.Replace("http","https")
+        $GalleryUrl = $url
+    }
+    catch
+    {
+        Write-Host "Failed to retrieve URL for testing"
+        Exit 1
+    }
+}
```
Comment (shishirx34) [inline tests/Scripts/Configure-FunctionalTests.ps1]: could you fix the indentation in this file.

File: tests/Scripts/Configure-FunctionalTests.ps1 (hunk 2 above)
Comment (scottbommarito) [inline tests/Scripts/Configure-FunctionalTests.ps1]: This file was based on a script from `TeamCity`. I would rather not change it.

File: tests/Scripts/RunP0Tests.bat [hunk 4]
```diff
@@ -36,14 +36,14 @@ if not "%errorlevel%"=="0" goto failure
 REM Run functional tests
 set testDir="NuGetGallery.FunctionalTests\bin\%config%"
 copy %nuget% %testDir%
-call %xunit% "%testDir%\NuGetGallery.FunctionalTests.dll" -trait "Category=%testCategory%" -teamcity -html functionaltestsP0.html
+call %xunit% "%testDir%\NuGetGallery.FunctionalTests.dll" -trait "Category=%testCategory%" -xml functionaltests.P0.xml
 
 REM Run web UI tests
-call %mstest% /TestContainer:"NuGetGallery.WebUITests.P0\bin\%config%\NuGetGallery.WebUITests.P0.dll" /TestSettings:Local.testsettings /detail:stdout /resultsfile:resultsfile.trx
+call %vstest% "NuGetGallery.WebUITests.P0\bin\%config%\NuGetGallery.WebUITests.P0.dll" /logger:trx
```
Comment (joelverhagen) [inline tests/Scripts/RunP0Tests.bat]: I think there's inconsistent line endings in this file.

Comment (joelverhagen) [inline tests/Scripts/RunP0Tests.bat]: Thus the `^M`.

Comment (scottbommarito) [inline tests/Scripts/RunP0Tests.bat]: Upon re-examining, it appears that the files had inconsistent line endings and these modifications fixed the issue.

Comment (joelverhagen) [inline tests/Scripts/RunP0Tests.bat]: Thanks for the investigation.

**Comment by scottbommarito** [comment]
@joelverhagen @shishirx34 @skofman1 Can you revisit `Configure-FunctionalTests.ps1` It's changed since your reviews

File: tests/Scripts/RunEverything.cmd [hunk 5]
```diff
@@ -13,9 +13,13 @@ REM Required Tools
 set msbuild="%PROGRAMFILES(X86)%\MsBuild\14.0\Bin\msbuild"
 set xunit="..\packages\xunit.runner.console.2.0.0\tools\xunit.console.exe"
```
Comment (joelverhagen) [inline tests/Scripts/RunEverything.cmd]: Same with other PR, can this be 2.1.0?

**Pull request merged**